src/main/java/org/sagebionetworks/bridge/dao/ParticipantDataDao.java

```diff
@@ -0,0 +1,58 @@
+package org.sagebionetworks.bridge.dao;
+
+import org.sagebionetworks.bridge.models.ForwardCursorPagedResourceList;
+import org.sagebionetworks.bridge.models.ParticipantData;
+
+public interface ParticipantDataDao {
+
+    /**
+     * Get participant data records for the given userId
+     * @param userId - the userId for the participant data
+     * @return list of participant data records in a resource list that includes original query values
+     */
+    ForwardCursorPagedResourceList<? extends ParticipantData> getParticipantData(String userId, String offsetKey, int pageSize);
+
+    /**
+     * Get participant data for the given userId
+     * @param userId
+     * @param offsetKey
+     * @param pageSize
+     * @return
+     */
+    ForwardCursorPagedResourceList<ParticipantData> getParticipantDataV4(String userId, String offsetKey, int pageSize);
+
+    //TODO check the functions below..
+
+    /**
+     * Get participant data record for the given userId and configId
+     * @param userId
+     * @param configId
+     * @param offsetKey
+     * @param pageSize
+     * @return
+     */
+    ParticipantData getParticipantDataRecordV4(String userId, String configId, String offsetKey, int pageSize);
+
+    /**
+     * Writes a participant data to the backing store.
+     * @param data - String data
+     */
+    void saveParticipantData(ParticipantData data); //TODO: how do we know the userId and configId of this data?
```

Comment:
Usually we add appId and userId in every method call (because we need both to load the user in a way that confirms this user is in the app context that the caller is signed into).